### PR TITLE
Feat save project

### PR DESCRIPTION
**Pull request by mixcas**
- Add route `/projects/add`
- Adds simple schema for validating added projects
- Adds view for adding projects with validation error handling

### CODE REVIEW

File: app/imports/components/projects/AddProject.jsx [hunk 1]
```diff
@@ -1,32 +1,80 @@
...
+      minDays: this.state.minDays,
+      maxDays: this.state.maxDays,
+      rate: this.state.rate,
+      currency: this.state.currency,
+      minQuote: this.state.minQuote,
+      maxQuote: this.state.maxQuote,
+    });
+  }
+
+  cleanObject(object) {
```
Comment (varffs) [inline app/imports/components/projects/AddProject.jsx:54]: should this be in some kind of more global scope?

File: app/imports/components/projects/AddProject.jsx [hunk 2]
```diff
@@ -45,18 +93,30 @@ export class AddProject extends Component {
...
-  onDaysChange(e) {
+  onMinDaysChange(e) {
+    // this.calculateTotalQuote is being passes as a callback that executes
+    // after the state mutation has happened
     this.setState({
-      days: e.target.value,
-      quote: e.target.value * this.state.rate,
-    });
+      minDays: parseInt(e.target.value),
+    }, this.calculateTotalQuote);
```
Comment (varffs) [inline app/imports/components/projects/AddProject.jsx:101]: is this a callback in the second variable?

**Review by varffs — COMMENTED** [review]
The obviously fails if you press the Add New Client button. But the clients and adding them will be another feature I guess. So just a few questions

File: app/imports/components/projects/AddProject.jsx (hunk 1 above)
Comment (mixcas) [reply app/imports/components/projects/AddProject.jsx:54]: yeah, maybe later

File: app/imports/components/projects/AddProject.jsx (hunk 2 above)
Comment (mixcas) [reply app/imports/components/projects/AddProject.jsx:101]: yup https://facebook.github.io/react/docs/react-component.html#setstate

**Pull request merged**